tests/genre.test.js

```diff
@@ -0,0 +1,132 @@
+//tests/genre.test.js
+const { expect } = require("chai");
+const request = require("supertest");
+const { Genre } = require("../src/models");
+const app = require("../src/app");
+
+describe("/genre", () => {
+  before(async () => Genre.sequelize.sync());
+
+  beforeEach(async () => {
+    await Genre.destroy({ where: {} });
+  });
+
+  describe("with no records in the database", () => {
+    describe("POST /genre", () => {
+      it("creates a new genre in the database", async () => {
+        const response = await request(app).post("/genre").send({
+          genre: "Crime & Thriller",
+        });
+        const newGenreRecord = await Genre.findByPk(response.body.id, {
+          raw: true,
+        });
+
+        expect(response.status).to.equal(201);
+        expect(newGenreRecord.genre).to.equal("Crime & Thriller");
+      });
+
+      it("checks validation when trying to create new book with empty string", async () => {
+        const response = await request(app).post("/genre").send({
+          genre: "",
+        });
+        const newGenreRecord = await Genre.findByPk(response.body.id, {
+          raw: true,
+        });
+
+        expect(response.status).to.equal(400);
+        expect(newGenreRecord).to.equal(null);
+      });
+    });
+  });
+
+  describe("with records in the database", () => {
+    let genres;
+
+    beforeEach(async () => {
+      genres = await Promise.all([
+        Genre.create({
+          genre: "Crime & Thriller",
+        }),
+        Genre.create({
+          genre: "Autobiography",
+        }),
+        Genre.create({
+          genre: "Sport",
+        }),
+      ]);
+    });
+
+    describe("GET /readers", () => {
+      it("gets all genre records", async () => {
+        const response = await request(app).get("/genre");
+
+        expect(response.status).to.equal(200);
+        expect(response.body.length).to.equal(3);
+
+        response.body.forEach((genre) => {
+          const expected = genres.find((a) => a.id === genre.id);
+
+          expect(genre.genre).to.equal(expected.genre);
+        });
```

Comment:
Variable renaming suggestion:
```suggestion
        response.body.forEach(responseGenre => {
          const expected = genres.find(testGenre => testGenre.id === responseGenre.id);

          expect(responseGenre.genre).to.equal(expected.genre);
        });
```